src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -228,6 +230,30 @@ def companion(order):
             reduced_companions.append(M)
         return (prod(reduced_companions) * companion_step * companion_initial)
 
+    @cached_method
+    def _frobenius_matrix_CSA(self):
+        r"""
+        Return the matrix of the left-multiplication by `\phi_T` with
+        respect to the maximal commutative subalgebra `K\{\tau^n\}` of
+        $K\{\tau\}$, where `K` is the base field and `n` is its degree
+        over `\mathbb F_q`. The characteristic polynomial of this matrix
+        is the characteristic polynomial of the Frobenius endomorphism,
+        which is crucial for :meth:`_frobenius_charpoly_CSA`.
+        """
+        E = self._base
+        EZ = PolynomialRing(E, name='Z')
+        n = self._base_degree_over_constants
+        phi_T = self.gen()
```

Comment:
I propose to replace `phi_T` by `f` because it will then be updated in the loop below.

Reply:
Done :+1: